### PR TITLE
Improve teachers flexibility when matching up students for #116

### LINKED ISSUE #116: Improve teachers flexibility when matching up students
Teachers should be able to quickly and easily shift around the unpaired students. They might want to pair a strong student with a weaker student etc.

Example improvements based on mock design below:
1) Different background colors to differentiate each pair
2) Add a pairup button so teachers can match just one pair at a time
3) Change the main pairing button text from "Pair up students" to "Pair up all students"

image

### PR DESCRIPTION
**Pull request by karlrez**
Resolves #116 

Added "Pair up" button, color for every other row, and modified text in "Pair up all students" button.
**Test plan:** Did a few test runs with 6 students, lots of pairing, unpairing, and shuffling, so far works for me. 

image

### CODE REVIEW

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 1]
```diff
@@ -118,27 +132,72 @@ export default function UnpairedStudentsList({
...
+          .map((pair, i) => (
+            <Grid
+              container
+              key={i}
+              sx={{
+                background: `${i % 2 === 0 ? '#f8e5e0' : ''}`,
+              }}
+            >
+              <Grid item xs={9}>
+                <Box>
```
Comment (mssiegel) [inline src/components/pages/TeachersPage/UnpairedStudentsList.tsx:150]: I think we can remove this Box that is wrapping it

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 2]
```diff
@@ -118,27 +132,72 @@ export default function UnpairedStudentsList({
...
+                  <UnpairedStudentItem
+                    i={i * 2}
+                    student={pair[0]}
+                    socket={socket}
+                    setUnpairedStudents={setUnpairedStudents}
+                  />
+                </Box>
+                <Box>
```
Comment (mssiegel) [inline src/components/pages/TeachersPage/UnpairedStudentsList.tsx:158]: I think we can remove this Box that is wrapping it

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 3]
```diff
@@ -118,27 +132,72 @@ export default function UnpairedStudentsList({
...
+              </Grid>
+
+              <Grid
+                item
+                xs={3}
+                sx={{
+                  textAlign: 'center',
+                }}
+              >
+                {pair[1] !== undefined && (
```
Comment (mssiegel) [inline src/components/pages/TeachersPage/UnpairedStudentsList.tsx:177]: We can remove the `!== undefined` and just check the value.  Also, I feel this `&&` check should be with the Grid item that wraps this button.

**Review by mssiegel — CHANGES_REQUESTED** [review]
I was really curious how we'd divide up the button to go on the right and this Grid system works really well! Great work.  I left a few minor comments to be addressed.

File: src/components/pages/TeachersPage/UnpairedStudentsList.tsx [hunk 4]
```diff
@@ -118,27 +132,72 @@ export default function UnpairedStudentsList({
...
-              setUnpairedStudents={setUnpairedStudents}
-            />
-          </Box>
-        ))}
+        {unpairedStudents
+          .reduce(function (studentPairs, currentValue, currentIndex, array) {
+            if (currentIndex % 2 === 0)
+              studentPairs.push(array.slice(currentIndex, currentIndex + 2));
+            return studentPairs;
+          }, [])
```
[the remark this answers is not in the document]
Comment (karlrez) [reply src/components/pages/TeachersPage/UnpairedStudentsList.tsx]: Yes! Using the chunk method is a much simplier way of accomplishing this.

**Review by mssiegel — APPROVED** [review]
Looks great to me!

**Pull request merged**